glusterd2/commands/volumes/common.go

```diff
@@ -28,16 +28,27 @@ const volumeIDXattrKey = "trusted.glusterfs.volume-id"
 
 // validateOptions validates if the options and their values are valid and can
 // be set on a volume.
-func validateOptions(opts map[string]string) error {
+func validateOptions(opts map[string]string, adv, exp, dep bool) error {
 
 	for k, v := range opts {
 		o, err := xlator.FindOption(k)
 		if err != nil {
 			return err
 		}
 
-		if !o.IsSettable() {
+		switch {
+		case !o.IsSettable():
 			return fmt.Errorf("Option %s cannot be set", k)
+
+		case o.IsAdvanced() && !adv:
```

Comment:
I thought the usecase was only to classify options into types to filter them out in listing and responses so that advanced options don't show up.

@nbalacha Isn't it ?

Reply:
It is needed for both. We need to filter out options in volume get and need to add an additional hurdle when setting.